src/Conversion/ONNXToMhlo/Tensor/ArgMax.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- ArgMax.cpp - Lowering ArgMax Op -------------------===//
+//
+// Copyright 2021-2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX ArgMax Operator to Mhlo dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToMhlo/ONNXToMhloCommon.hpp"
+#include "src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.hpp"
+#include "src/Support/TypeUtilities.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+static void BuildArgmaxReductionBody(
+    Type elementType, Type indexElementType, Region *body, OpBuilder *builder) {
+  OpBuilder::InsertionGuard guard(*builder);
+  Type inputType = RankedTensorType::get(/*shape=*/{}, elementType);
+  Type indexType = RankedTensorType::get(/*shape=*/{}, indexElementType);
+  Block *block = builder->createBlock(body);
+  Location loc = body->getLoc();
+  block->addArguments({inputType, indexType, inputType, indexType},
+      SmallVector<Location, 4>(4, loc));
+
+  Value lhsVal = block->getArgument(0);
+  Value lhsIndex = block->getArgument(1);
+  Value rhsVal = block->getArgument(2);
+  Value rhsIndex = block->getArgument(3);
+
+  ImplicitLocOpBuilder b(loc, *builder);
+  Value compareDt =
+      b.create<mhlo::CompareOp>(lhsVal, rhsVal, mhlo::ComparisonDirection::GE);
+  Value selectedInput =
+      b.create<mhlo::SelectOp>(inputType, compareDt, lhsVal, rhsVal);
+
+  Value compareEq =
+      b.create<mhlo::CompareOp>(lhsVal, rhsVal, mhlo::ComparisonDirection::EQ);
+  Value minIndex = b.create<mhlo::MinOp>(lhsIndex, rhsIndex);
+  Value minValIndex =
+      b.create<mhlo::SelectOp>(indexType, compareDt, lhsIndex, rhsIndex);
+  Value selectedIndex =
+      b.create<mhlo::SelectOp>(indexType, compareEq, minIndex, minValIndex);
+  Value returnValues[] = {selectedInput, selectedIndex};
+  b.create<mhlo::ReturnOp>(returnValues);
+}
+
+struct ONNXArgMaxOpLoweringToMhlo : public ConversionPattern {
+  ONNXArgMaxOpLoweringToMhlo(MLIRContext *ctx)
+      : ConversionPattern(mlir::ONNXArgMaxOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    // Gather info.
+    Location loc = op->getLoc();
+    ONNXArgMaxOpAdaptor operandAdaptor(operands);
+    ONNXArgMaxOp argMaxOp = llvm::cast<ONNXArgMaxOp>(op);
+
+    // shape helper
+    ONNXArgMaxOpShapeHelper shapeHelper(&argMaxOp);
+    LogicalResult shapecomputed = shapeHelper.computeShape(operandAdaptor);
+    (void)shapecomputed;
+    assert(!failed(shapecomputed) && "expected to succeed");
```

Comment:
message is not very informative, I think "Could not compute output shape" would be better

Reply:
will fix that